pymc4/forward_sampling.py

```diff
@@ -112,24 +109,24 @@ def sample_prior_predictive(
     >>> prior_samples = sample_prior_predictive(
     ...     model2(), sample_shape=(20,), sample_from_observed=False
     ... )
-    >>> np.allclose(np.mean(prior_samples["model2/y"]), 1)
+    >>> np.allclose(np.mean(prior_samples.prior_predictive["model2/y"]), 1)
     True
 
     Furthermore, this has consequences at the shape level of the drawn samples
     >>> prior_samples["model2/y"].shape
-    (20, 10)
+    (20, 20)
 
-    If ``sample_from_observed=True`` the value of the ``x`` random variable
-    will be drawn from its prior distribution, which will have consequences
-    both at the value and shape levels of downstream computations
+    If ``sample_from_observed=True`` the value of the ``x`` random variable will be drawn from its
+    prior distribution, which will have consequences both at the value and shape levels of
+    downstream computations
 
     >>> prior_samples = sample_prior_predictive(
     ...     model2(), sample_shape=(20,), sample_from_observed=True
     ... )
     >>> np.allclose(np.mean(prior_samples["model2/y"]), 1)
     False
     >>> prior_samples["model2/y"].shape
-    (20,)
+    (1, 20)
```

Comment:
I think this is because the first dimension is the number of chains.

Reply:
Does arviz impose the notion of chains in prior predictive sampling? I think this can be very dangerous in models with many random variables, some that are scalar and some that are multidimensional. The scalar ones might end up with an extra "chain" axis, but what happens to the multidimensional ones? Do they get an extra axis as well or are they left as they were, trusting that some sample shape axis indicated the chain membership? In any case, it will lead to potential and unexpected broadcasting problems down the line.

Reply:
arviz assumes you always indicate the number of chains, and that it is always a single scalar.

Reply:
I understand that the chain dimension comes out naturally from `sample` and thus it makes its way into the posterior predictive samples. But I think that it isn't right to impose it to the prior predictive samples.
Furthermore, one cannot make assumptions on whether it is correct to always add a dimension to the samples from the prior. For example, in smc with multiple chains, one has to explicitly draw independent samples from the prior for each chain. It wouldn't be right to later add an extra dimension to the resulting samples.